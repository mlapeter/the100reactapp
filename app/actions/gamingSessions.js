export const CREATE_GAMING_SESSION = "CREATE_GAMING_SESSION";
export const CREATE_GAMING_SESSION_RESULT =
  "CREATE_GAMING_SESSION_ERROR_RESULT";
export const CREATE_GAMING_SESSION_ERROR = "CREATE_GAMING_SESSION_ERROR";

export const EDIT_GAMING_SESSION = "EDIT_GAMING_SESSION";
export const EDIT_GAMING_SESSION_RESULT = "EDIT_GAMING_SESSION_RESULT";
export const EDIT_GAMING_SESSION_ERROR = "EDIT_GAMING_SESSION_ERROR";

export const DELETE_GAMING_SESSION = "DELETE_GAMING_SESSION";
export const DELETE_GAMING_SESSION_RESULT = "DELETE_GAMING_SESSION_RESULT";
export const DELETE_GAMING_SESSION_ERROR = "DELETE_GAMING_SESSION_ERROR";

export const FETCH_GAMING_SESSION = "FETCH_GAMING_SESSION";
export const FETCH_GAMING_SESSION_RESULT = "FETCH_GAMING_SESSION_RESULT";
export const FETCH_GAMING_SESSION_ERROR = "FETCH_GAMING_SESSION_ERROR";

export const FETCH_GAMING_SESSIONS = "FETCH_GAMING_SESSIONS";
export const FETCH_GAMING_SESSIONS_RESULT = "FETCH_GAMING_SESSIONS_RESULT";
export const FETCH_GAMING_SESSIONS_ERROR = "FETCH_GAMING_SESSIONS_ERROR";
export const FETCH_GAMING_SESSIONS_NO_DATA = "FETCH_GAMING_SESSIONS_NO_DATA";
export const REFRESH_GAMING_SESSIONS = "REFRESH_GAMING_SESSIONS";
export const LOAD_MORE_GAMING_SESSIONS = "LOAD_MORE_GAMING_SESSIONS";
export const LOAD_MORE_GAMING_SESSIONS_RESULT =
  "LOAD_MORE_GAMING_SESSIONS_RESULT";

export const FETCH_MY_GAMING_SESSIONS = "FETCH_MY_GAMING_SESSIONS";
export const FETCH_MY_GAMING_SESSIONS_RESULT =
  "FETCH_MY_GAMING_SESSIONS_RESULT";
export const FETCH_MY_GAMING_SESSIONS_ERROR = "FETCH_MY_GAMING_SESSIONS_ERROR";
export const FETCH_MY_GAMING_SESSIONS_NO_DATA =
  "FETCH_MY_GAMING_SESSIONS_NO_DATA";
export const CLEAR_MY_GAMING_SESSIONS = "ClEAR_MY_GAMING_SESSIONS";
export const REFRESH_MY_GAMING_SESSIONS = "REFRESH_MY_GAMING_SESSIONS";
export const LOAD_MORE_MY_GAMING_SESSIONS = "LOAD_MORE_MY_GAMING_SESSIONS";
export const LOAD_MORE_MY_GAMING_SESSIONS_RESULT =
  "LOAD_MORE_MY_GAMING_SESSIONS_RESULT";

export const FETCH_GROUP_GAMING_SESSIONS = "FETCH_GROUP_GAMING_SESSIONS";
export const FETCH_GROUP_GAMING_SESSIONS_RESULT =
  "FETCH_GROUP_GAMING_SESSIONS_RESULT";
export const FETCH_GROUP_GAMING_SESSIONS_ERROR =
  "FETCH_GROUP_GAMING_SESSIONS_ERROR";
export const FETCH_GROUP_GAMING_SESSIONS_NO_DATA =
  "FETCH_GROUP_GAMING_SESSIONS_NO_DATA";

export const CLEAR_GROUP_GAMING_SESSIONS = "CLEAR_GROUP_GAMING_SESSIONS";

export const REFRESH_GROUP_GAMING_SESSIONS = "REFRESH_GROUP_GAMING_SESSIONS";
export const LOAD_MORE_GROUP_GAMING_SESSIONS =
  "LOAD_MORE_GROUP_GAMING_SESSIONS";
export const LOAD_MORE_GROUP_GAMING_SESSIONS_RESULT =
  "LOAD_MORE_GROUP_GAMING_SESSIONS_RESULT";

export const createGamingSession = gamingSession => ({
  type: CREATE_GAMING_SESSION,
  gamingSession
});

export const editGamingSession = gamingSession => ({
  type: EDIT_GAMING_SESSION,
  gamingSession
});

export const deleteGamingSession = deleteGamingSessionId => ({
  type: DELETE_GAMING_SESSION,
  deleteGamingSessionId
});

export const fetchGamingSession = gamingSessionId => ({
  type: FETCH_GAMING_SESSION,
  gamingSessionId
});

export const fetchGamingSessions = endpoint => ({
  type: FETCH_GAMING_SESSIONS,
  endpoint
});

export const refreshGamingSessions = endpoint => ({
  type: REFRESH_GAMING_SESSIONS,
  endpoint
});

export const loadMoreGamingSessions = endpoint => ({
  type: LOAD_MORE_GAMING_SESSIONS,
  endpoint
});

export const fetchMyGamingSessions = endpoint => ({
  type: FETCH_MY_GAMING_SESSIONS,
  endpoint
});

export const refreshMyGamingSessions = endpoint => ({
  type: REFRESH_MY_GAMING_SESSIONS,
  endpoint
});

export const loadMoreMyGamingSessions = endpoint => ({
  type: LOAD_MORE_MY_GAMING_SESSIONS,
  endpoint
});

export const fetchGroupGamingSessions = endpoint => ({
  type: FETCH_GROUP_GAMING_SESSIONS,
  endpoint
});

export const refreshGroupGamingSessions = endpoint => ({
  type: REFRESH_GROUP_GAMING_SESSIONS,
  endpoint
});

export const loadMoreGroupGamingSessions = endpoint => ({
  type: LOAD_MORE_GROUP_GAMING_SESSIONS,
  endpoint
});
