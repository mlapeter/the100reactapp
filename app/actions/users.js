export const UPDATE_USER = "UPDATE_USER";
export const UPDATE_USER_RESULT = "UPDATE_USER_RESULT";
export const UPDATE_USER_ERROR = "UPDATE_USER_ERROR";

export const FETCH_USER = "FETCH_USER";
export const FETCH_USER_RESULT = "FETCH_USER_RESULT";
export const FETCH_USER_ERROR = "FETCH_USER_ERROR";

export const FETCH_CURRENT_USER = "FETCH_CURRENT_USER";
export const FETCH_CURRENT_USER_RESULT = "FETCH_CURRENT_USER_RESULT";
export const CLEAR_CURRENT_USER = "CLEAR_CURRENT_USER";

export const FETCH_FRIENDS = "FETCH_FRIENDS";
export const FETCH_FRIENDS_RESULT = "FETCH_FRIENDS_RESULT";
export const FETCH_FRIENDS_ERROR = "FETCH_FRIENDS_ERROR";
export const FETCH_FRIENDS_NO_DATA = "FETCH_FRIENDS_NO_DATA";
export const LOAD_MORE_FRIENDS = "LOAD_MORE_FRIENDS";
export const LOAD_MORE_FRIENDS_RESULT = "LOAD_MORE_FRIENDS_RESULT";
export const CHANGE_FRIENDS_PAGE = "CHANGE_FRIENDS_PAGE";
export const REFRESH_FRIENDS = "REFRESH_FRIENDS";

export const FETCH_PENDING_FRIENDS = "FETCH_PENDING_FRIENDS";
export const FETCH_PENDING_FRIENDS_RESULT = "FETCH_PENDING_FRIENDS_RESULT";
export const FETCH_PENDING_FRIENDS_ERROR = "FETCH_PENDING_FRIENDS_ERROR";
export const FETCH_PENDING_FRIENDS_NO_DATA = "FETCH_PENDING_FRIENDS_NO_DATA";
export const LOAD_MORE_PENDING_FRIENDS = "LOAD_MORE_PENDING_FRIENDS";
export const LOAD_MORE_PENDING_FRIENDS_RESULT =
  "LOAD_MORE_PENDING_FRIENDS_RESULT";

export const CHANGE_PENDING_FRIENDS_PAGE = "CHANGE_PENDING_FRIENDS_PAGE";
export const REFRESH_PENDING_FRIENDS = "REFRESH_PENDING_FRIENDS";

export const FETCH_GROUP_MEMBERS = "FETCH_GROUP_MEMBERS";
export const FETCH_GROUP_MEMBERS_RESULT = "FETCH_GROUP_MEMBERS_RESULT";
export const FETCH_GROUP_MEMBERS_ERROR = "FETCH_GROUP_MEMBERS_ERROR";
export const FETCH_GROUP_MEMBERS_NO_DATA = "FETCH_GROUP_MEMBERS_NO_DATA";
export const LOAD_MORE_GROUP_MEMBERS = "LOAD_MORE_GROUP_MEMBERS";
export const LOAD_MORE_GROUP_MEMBERS_RESULT = "LOAD_MORE_GROUP_MEMBERS_RESULT";
export const CHANGE_GROUP_MEMBERS_PAGE = "CHANGE_GROUP_MEMBERS_PAGE";
export const REFRESH_GROUP_MEMBERS = "REFRESH_GROUP_MEMBERS";

export const fetchUser = userId => ({
  type: FETCH_USER,
  userId
});

export const fetchCurrentUser = () => ({
  type: FETCH_CURRENT_USER
});

export const updateUser = currentUser => ({
  type: UPDATE_USER,
  currentUser
});

export const clearCurrentUser = () => ({
  type: CLEAR_CURRENT_USER
});

export const fetchFriends = () => ({
  type: FETCH_FRIENDS
});

export const loadMoreFriends = () => ({
  type: LOAD_MORE_FRIENDS
});

export const changeFriendsPage = () => ({
  type: CHANGE_FRIENDS_PAGE
});

export const refreshFriends = () => ({
  type: REFRESH_FRIENDS
});

export const fetchPendingFriends = () => ({
  type: FETCH_PENDING_FRIENDS
});

export const loadMorePendingFriends = () => ({
  type: LOAD_MORE_PENDING_FRIENDS
});

export const refreshPendingFriends = () => ({
  type: REFRESH_PENDING_FRIENDS
});

export const fetchGroupMembers = () => ({
  type: FETCH_GROUP_MEMBERS
});

export const loadMoreGroupMembers = () => ({
  type: LOAD_MORE_GROUP_MEMBERS
});

export const refreshGroupMembers = () => ({
  type: REFRESH_GROUP_MEMBERS
});
