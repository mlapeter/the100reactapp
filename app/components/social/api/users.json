[
    {
        "id": "derek.russel",
        "name": "Derek Russel",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/8bbac5ad06b4a569b8a446825f7371c81ebac821.png?alt=media&token=73b33332-c587-464b-af68-52554221b73a",
        "caption": ""
    },
    {
        "id": "jmitch",
        "name": "Joe Mitchell",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/9b5cb1a55d786aebebfe7379dc16b8ccf0e81942.png?alt=media&token=6f3c5b07-1b67-478b-9b46-15cfee8d6f54",
        "caption": ""
    },
    {
        "id": "monicaa",
        "name": "Monica Dixon",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/a71de5589d604ece4a685c2c270267cebe192be8.png?alt=media&token=0d91856a-8a43-4b58-8e44-57becc3f34eb",
        "caption": ""
    },
    {
        "id": "andrea.schmidt",
        "name": "Andrea Schmidt",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/6e37e1c924aa55072f25e47ede573285a6fb69bf.png?alt=media&token=ae4e9186-91b8-4748-963a-02138a3e395c",
        "caption": ""
    },
    {
        "id": "alexandergarcia",
        "name": "Alex Garcia",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/a2c02707b8423da62b591e7111e790a77917cf14.png?alt=media&token=98e89375-36ce-4477-b19a-7d68194de3e8",
        "caption": ""
    },
    {
        "id": "mjcole",
        "name": "Michael Cole",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/40eabfbf1c4cd057304d9a18e260d42c7025b5ca.png?alt=media&token=ebbc0a41-61db-4bed-a0c5-8f088eb16c21",
        "caption": ""
    },
    {
        "id": "schavez",
        "name": "Sylvia Chavez",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/6e37e1c924aa55072f25e47ede573285a6fb69bf.png?alt=media&token=ae4e9186-91b8-4748-963a-02138a3e395c",
        "caption": "Aspiring creative writer. I love ðŸŒ¶ food and good people.",
        "cover": {
            "uri":  "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/e88d0539c04f38d3b7f7d030779f11feb26c3660.png?alt=media&token=7d2041fa-ed1f-41ee-ad6a-8af1531d657b",
            "preview": "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAoAAAAICAYAAADA+m62AAAAAXNSR0IArs4c6QAAACBjSFJNAAB6JgAAgIQAAPoAAACA6AAAdTAAAOpgAAA6mAAAF3CculE8AAAACXBIWXMAAAsTAAALEwEAmpwYAAABWWlUWHRYTUw6Y29tLmFkb2JlLnhtcAAAAAAAPHg6eG1wbWV0YSB4bWxuczp4PSJhZG9iZTpuczptZXRhLyIgeDp4bXB0az0iWE1QIENvcmUgNS40LjAiPgogICA8cmRmOlJERiB4bWxuczpyZGY9Imh0dHA6Ly93d3cudzMub3JnLzE5OTkvMDIvMjItcmRmLXN5bnRheC1ucyMiPgogICAgICA8cmRmOkRlc2NyaXB0aW9uIHJkZjphYm91dD0iIgogICAgICAgICAgICB4bWxuczp0aWZmPSJodHRwOi8vbnMuYWRvYmUuY29tL3RpZmYvMS4wLyI+CiAgICAgICAgIDx0aWZmOk9yaWVudGF0aW9uPjE8L3RpZmY6T3JpZW50YXRpb24+CiAgICAgIDwvcmRmOkRlc2NyaXB0aW9uPgogICA8L3JkZjpSREY+CjwveDp4bXBtZXRhPgpMwidZAAAA8ElEQVQYGRWPS1LCQABEX2YmwyR8EmRheRHv4Vm8rUstsQoKMAbI/Mew7a5+3V29vr+VKxa5KmSRmI6J4gSls6QQSHvB/X5H9M8Na9MQfuB8vlG2Gd0I5K+EKpONx/95VJSebqtnsWC/J1x50CFOkTBm4tpDn1Ff+4FWStKiUK0q4ikwVhEnA3FISJ9JtUe8mJ48FcIlUhfY7TYsQv2oousNEoWUGuWspVE1q7wgztRjHphyoNdLyrzoGiY27RJxGE/E+Z3QEpcd++FI6yW6NVxxSAvqpmZ/junaYHXg4/RJf1Y8dVucCUyHEWMNfj72D6rqgcOmcLiwAAAAAElFTkSuQmCC"
        }
    },
    {
        "id": "griffinsof",
        "name": "Sofia Griffin",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/8d8edc4c628969721a3127824566bcf7bef353c9.png?alt=media&token=99e98755-78ec-439b-ac51-778cba2d2cda",
        "caption": ""
    },
    {
        "id": "itsmesuri",
        "name": "Suri H.",
        "picture": "https://firebasestorage.googleapis.com/v0/b/react-native-e.appspot.com/o/881c9bfb4e5d3578355f328be01fe1d5152bbb27.png?alt=media&token=5ebcebcd-0c90-4d9d-8c45-daa05e6f1f41",
        "caption": ""
    }
]
