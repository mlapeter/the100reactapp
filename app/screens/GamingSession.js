import React, { Component } from "react";
import {
  ActivityIndicator,
  AsyncStorage,
  LayoutAnimation,
  Platform,
  Share,
  StyleSheet,
  Text,
  View,
  Vibration
} from "react-native";
import { Analytics, PageHit } from "expo-analytics";
import Environment from "../config/environment";

import { colors, fontSizes, fontStyles, styleSheet } from "../styles";
import { connect } from "react-redux";
import { connectAlert } from "../components/Alert";
import {
  fetchGamingSession,
  fetchMyGamingSessions,
  fetchGroupGamingSessions
} from "../actions/gamingSessions";
import ChatPreview from "../components/ChatPreview";
import Panel from "../components/Panel/Panel";
import PlayersList from "../components/PlayersList";
import Header from "../components/Header";
import Content from "../components/Content";
import Card from "../components/Card";
import NavigationBar from "../components/NavigationBar";
import GamingSessionIconBar from "../components/GamingSessionIconBar";
import GroupsList from "../components/GroupsList";
import { fetchBungieImage } from "../utils/destinyActivities";

class GamingSession extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      reserveButtonVisible: false,
      isLoading: true,
      gamingSessionErrorAt: null
    };
    gamingSessionId = this.props.navigation.state.params.gamingSessionId;
  }

  componentWillMount() {
    this.loadingTimer = setTimeout(() => {
      this.setState({
        isLoading: false
      });
    }, 800);
  }

  componentWillUnmount() {
    console.log("UNMOUNTING GAMING SESSION");
    if (this.loadingTimer) {
      clearTimeout(this.loadingTimer);
    }
    if (this.updateTimer) {
      clearTimeout(this.updateTimer);
    }
  }

  componentDidMount() {
    this.fetchGamingSessionData();
    const analytics = new Analytics(Environment["GOOGLE_ANALYTICS_ID"]);
    analytics
      .hit(new PageHit("App - Gaming Session"))
      .catch(e => console.log(e.message));
  }

  fetchGamingSessionData() {
    this.props.dispatch(fetchGamingSession(gamingSessionId));
  }

  joinGame = () => {
    this.postData("/join");
  };

  joinGameAsReserve = () => {
    this.postData("/join?join_as_reserve=true");
    this.onLongPress();
  };

  leaveGame = () => {
    this.postData("/leave");
  };

  onLongPress = () => {
    Vibration.vibrate(20);
    this.setState({
      reserveButtonVisible: !this.state.reserveButtonVisible
    });
    LayoutAnimation.configureNext(LayoutAnimation.Presets.spring);
  };

  postData(action) {
    this.setState({
      isLoading: true
    });
    AsyncStorage.getItem("id_token").then(token => {
      fetch(
        Environment["API_BASE_URL"] +
          Environment["API_VERSION"] +
          "gaming_sessions/" +
          gamingSessionId +
          action,
        {
          method: "POST",
          headers: {
            "Content-Type": "application/json",
            Authorization: "Bearer " + token
          }
        }
      )
        .then(response => response.json())
        .then(responseJson => {
          if (action === "/join" || action === "/join?join_as_reserve=true") {
            this.fetchGamingSessionData();
          } else {
            this.props.navigation.navigate("GamingSessionsList");
          }
          this.updateTimer = setTimeout(() => {
            this.props.dispatch(fetchMyGamingSessions());
            this.props.dispatch(fetchGroupGamingSessions());
            this.setState({
              isLoading: false
            });
          }, 300);
        })
        .catch(error => {
          console.log(error);
        });
    });
  }

  onShare() {
    Share.share(
      {
        message:
          Platform.OS === "android"
            ? "Join my game: " +
              this.props.gamingSession.category.toString() +
              " " +
              "https://the100.io/game/" +
              this.props.gamingSession.id
            : "Join my game: " +
              this.props.gamingSession.category.toString() +
              " ",
        url: "https://the100.io/game/" + this.props.gamingSession.id,
        title: ""
      },
      {
        // Android only:
        dialogTitle: "Share Gaming Session"
      }
    );
  }

  render() {
    const { params } = this.props.navigation.state;
    const navigation = this.props.navigation;

    if (
      this.state.isLoading ||
      this.props.gamingSessionLoading ||
      !this.props.gamingSession ||
      !this.props.gamingSession.confirmed_sessions ||
      !this.props.user
    ) {
      return (
        <View style={styles.container}>
          <Header
            title={"..."}
            picture={"img/default-gaming-session-header.jpg"}
            heightRatio={0.5}
            topGradientTransparency={"rgba(0,0,0,0.9)"}
            middleGradientTransparency={"rgba(0,0,0,0.9)"}
            bottomGradientTransparency={"rgba(0,0,0,0.9)"}
          >
            <NavigationBar
              back="Games"
              type="transparent"
              {...{ navigation }}
            />
          </Header>
          <Content>
            <ActivityIndicator style={styles.loading} />
          </Content>
        </View>
      );
    }

    let userIds = [];
    this.props.gamingSession.confirmed_sessions.map(confirmedSession =>
      userIds.push(confirmedSession.user_id)
    );

    const rightAction =
      this.state.reserveButtonVisible === true
        ? null
        : {
            icon: "share",
            size: 24,
            onPress: () => {
              this.onShare();
            }
          };
    const rightAction2 = userIds.includes(this.props.user.id)
      ? {
          icon: "cancel",
          text: "Leave",
          size: 24,
          onPress: () => {
            this.leaveGame();
          }
        }
      : {
          icon: "outline-person_add-24px",
          text: "Join",
          size: 24,
          onPress: () => {
            this.joinGame();
          },
          onLongPress: () => {
            this.onLongPress();
          }
        };
    const rightAction3 =
      userIds.includes(this.props.user.id) &&
      this.props.user.id === this.props.gamingSession.creator_id
        ? {
            icon: "edit",
            text: "Edit",
            size: 24,
            onPress: () => {
              this.props.navigation.navigate("GamingSessionEdit");
            }
          }
        : this.state.reserveButtonVisible === true
          ? {
              icon: "outline-person_add-24px",
              text: "Join as Reserve",
              size: 24,
              onPress: () => {
                this.joinGameAsReserve();
              },
              onLongPress: () => {
                this.onLongPress();
              }
            }
          : null;

    let room = `game-${this.props.gamingSession.id}`;
    let url = `chat/gaming_sessions/${room}`;

    return (
      <View style={styles.container}>
        <Header
          title={this.props.gamingSession.category}
          picture={fetchBungieImage(this.props.gamingSession.category)}
          heightRatio={0.5}
          topGradientTransparency={"rgba(0,0,0,0.4)"}
          middleGradientTransparency={"rgba(0,0,0,0.1)"}
          bottomGradientTransparency={"rgba(0,0,0,0.6)"}
        >
          <NavigationBar
            back="Games"
            type="transparent"
            {...{ navigation, rightAction, rightAction2, rightAction3 }}
          />
        </Header>
        <GamingSessionIconBar
          startTime={this.props.gamingSession.start_time}
          platform={this.props.gamingSession.platform}
          primaryUsersCount={this.props.gamingSession.primary_users_count}
          teamSize={this.props.gamingSession.team_size}
          lightLevel={this.props.gamingSession.light_level}
          sherpaLed={this.props.gamingSession.sherpa_led}
        />
        <Content>
          {this.props.gamingSession.name ? (
            <Card>
              <Panel text={this.props.gamingSession.name} numberOfLines={5} />
            </Card>
          ) : null}
          <Card>
            <PlayersList
              confirmedSessions={this.props.gamingSession.confirmed_sessions}
              navigation={this.props.navigation}
              platform={this.props.gamingSession.platform}
            />
          </Card>
          <Card
            onPress={() =>
              this.props.navigation.navigate("GamingSessionChat", {
                title: "Gaming Session Chat",
                room: room,
                url: url,
                allowAnon: true
              })
            }
          >
            <Text style={[styles.headline, styleSheet.typography["headline"]]}>
              Player Chat &raquo;
            </Text>
            <ChatPreview
              room={room}
              url={url}
              allowAnon={true}
              onOpenChat={() =>
                this.props.navigation.navigate("GamingSessionChat", {
                  title: "Gaming Session Chat",
                  room: room,
                  url: url,
                  allowAnon: true
                })
              }
            />
          </Card>
          {!this.props.gamingSession.group_id ||
          !this.props.gamingSession.group ? null : (
            <GroupsList
              groups={[this.props.gamingSession.group]}
              navigation={this.props.navigation}
            />
          )}
        </Content>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1
  },
  loading: {
    flex: 1,
    padding: 20,
    alignItems: "center",
    justifyContent: "center"
  }
});

const mapStateToProps = state => {
  const user = state.users.currentUser;
  const gamingSessionLoading = state.gamingSessions.gamingSessionLoading;
  const gamingSession = state.gamingSessions.gamingSession;
  const gamingSessionsError = state.gamingSessions.error;
  const gamingSessionsErrorAt = state.gamingSessions.errorAt;

  return {
    user,
    gamingSessionLoading,
    gamingSession,
    gamingSessionsError,
    gamingSessionsErrorAt
  };
};

export default connect(mapStateToProps)(connectAlert(GamingSession));
