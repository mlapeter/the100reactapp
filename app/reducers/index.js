import { combineReducers } from "redux";

import authentication from "./authentication";
import search from "./search";
import notifications from "./notifications";
import users from "./users";
import group from "./group";
import gamingSessions from "./gamingSessions";
import onboarding from "./onboarding";
import conversations from "./conversations";
import feed from "./feed";

export default combineReducers({
  authentication,
  search,
  notifications,
  users,
  group,
  gamingSessions,
  onboarding,
  conversations,
  feed
});
