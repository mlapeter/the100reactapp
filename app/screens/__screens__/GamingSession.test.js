import React from "react";
import { shallow } from "enzyme";
import configureStore from "redux-mock-store";

import GamingSession from "../GamingSession";
import { initialState as gamingSessionsInitialState } from "../../reducers/gamingSessions";
import { initialState as usersInitialState } from "../../reducers/users";

const mockStore = configureStore([]);

it("successfully renders", () => {
  const navigation = { state: { params: {} } };
  const initialState = {
    gamingSessions: gamingSessionsInitialState,
    users: usersInitialState
  };
  const rendered = shallow(<GamingSession navigation={navigation} />, {
    context: { store: mockStore(initialState) }
  });
  expect(rendered.dive().dive()).toMatchSnapshot();
});
