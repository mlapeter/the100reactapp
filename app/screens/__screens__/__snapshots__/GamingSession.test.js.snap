// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`successfully renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GamingSession
    dispatch={[Function]}
    gamingSession={Object {}}
    gamingSessionLoading={true}
    navigation={
      Object {
        "state": Object {
          "params": Object {},
        },
      }
    }
    user={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <Header
          bottomGradientTransparency="rgba(0,0,0,0.9)"
          heightRatio={0.5}
          middleGradientTransparency="rgba(0,0,0,0.9)"
          picture="img/default-gaming-session-header.jpg"
          title="..."
          topGradientTransparency="rgba(0,0,0,0.9)"
        >
          <NavigationBar
            back="Games"
            color="white"
            expanded={false}
            navigation={
              Object {
                "state": Object {
                  "params": Object {},
                },
              }
            }
            title=""
            type="transparent"
            withGradient={false}
          />
        </Header>,
        <Content>
          <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "padding": 20,
              }
            }
          />
        </Content>,
      ],
      "style": Object {
        "flex": 1,
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bottomGradientTransparency": "rgba(0,0,0,0.9)",
          "children": <NavigationBar
            back="Games"
            color="white"
            expanded={false}
            navigation={
              Object {
                "state": Object {
                  "params": Object {},
                },
              }
            }
            title=""
            type="transparent"
            withGradient={false}
          />,
          "heightRatio": 0.5,
          "middleGradientTransparency": "rgba(0,0,0,0.9)",
          "picture": "img/default-gaming-session-header.jpg",
          "title": "...",
          "topGradientTransparency": "rgba(0,0,0,0.9)",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "back": "Games",
            "color": "white",
            "expanded": false,
            "navigation": Object {
              "state": Object {
                "params": Object {},
              },
            },
            "title": "",
            "type": "transparent",
            "withGradient": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <ActivityIndicator
            animating={true}
            color="#999999"
            hidesWhenStopped={true}
            size="small"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "padding": 20,
              }
            }
          />,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "animating": true,
            "color": "#999999",
            "hidesWhenStopped": true,
            "size": "small",
            "style": Object {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
              "padding": 20,
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <Header
            bottomGradientTransparency="rgba(0,0,0,0.9)"
            heightRatio={0.5}
            middleGradientTransparency="rgba(0,0,0,0.9)"
            picture="img/default-gaming-session-header.jpg"
            title="..."
            topGradientTransparency="rgba(0,0,0,0.9)"
          >
            <NavigationBar
              back="Games"
              color="white"
              expanded={false}
              navigation={
                Object {
                  "state": Object {
                    "params": Object {},
                  },
                }
              }
              title=""
              type="transparent"
              withGradient={false}
            />
          </Header>,
          <Content>
            <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="small"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "padding": 20,
                }
              }
            />
          </Content>,
        ],
        "style": Object {
          "flex": 1,
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bottomGradientTransparency": "rgba(0,0,0,0.9)",
            "children": <NavigationBar
              back="Games"
              color="white"
              expanded={false}
              navigation={
                Object {
                  "state": Object {
                    "params": Object {},
                  },
                }
              }
              title=""
              type="transparent"
              withGradient={false}
            />,
            "heightRatio": 0.5,
            "middleGradientTransparency": "rgba(0,0,0,0.9)",
            "picture": "img/default-gaming-session-header.jpg",
            "title": "...",
            "topGradientTransparency": "rgba(0,0,0,0.9)",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "back": "Games",
              "color": "white",
              "expanded": false,
              "navigation": Object {
                "state": Object {
                  "params": Object {},
                },
              },
              "title": "",
              "type": "transparent",
              "withGradient": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <ActivityIndicator
              animating={true}
              color="#999999"
              hidesWhenStopped={true}
              size="small"
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "justifyContent": "center",
                  "padding": 20,
                }
              }
            />,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "animating": true,
              "color": "#999999",
              "hidesWhenStopped": true,
              "size": "small",
              "style": Object {
                "alignItems": "center",
                "flex": 1,
                "justifyContent": "center",
                "padding": 20,
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "context": Object {
      "store": Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      },
    },
  },
}
`;
